src/OrbitGl/CGroupAndProcessMemoryTrack.cpp

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "CGroupAndProcessMemoryTrack.h"
+
+#include <absl/strings/str_format.h>
+#include <absl/strings/substitute.h>
+
+#include "DisplayFormats/DisplayFormats.h"
+
+namespace orbit_gl {
+
+namespace {
+
+const std::string kTrackValueLabelUnit = "MB";
+const std::string kTrackName = absl::StrFormat("CGroup Memory Usage (%s)", kTrackValueLabelUnit);
+
+static std::array<std::string, kCGroupAndProcessMemoryTrackDimension> CreateSeriesName(
+    const std::string& cgroup_name, const std::string& process_name) {
+  return {absl::StrFormat("Process [%s] RssAnon", process_name), "Other Processes RssAnon",
+          absl::StrFormat("CGroup [%s] Mapped File", cgroup_name),
+          absl::StrFormat("CGroup [%s] Unused", cgroup_name)};
+}
+
+}  // namespace
+
+CGroupAndProcessMemoryTrack::CGroupAndProcessMemoryTrack(
+    CaptureViewElement* parent, TimeGraph* time_graph, orbit_gl::Viewport* viewport,
+    TimeGraphLayout* layout, const std::string& cgroup_name,
+    const orbit_client_model::CaptureData* capture_data)
+    : MemoryTrack<kCGroupAndProcessMemoryTrackDimension>(
+          parent, time_graph, viewport, layout, kTrackName,
+          CreateSeriesName(cgroup_name, capture_data->process_name()), capture_data),
+      cgroup_name_(cgroup_name) {
+  SetLabelUnit(kTrackValueLabelUnit);
+
+  constexpr uint8_t kTrackValueDecimalDigits = 2;
+  SetNumberOfDecimalDigits(kTrackValueDecimalDigits);
+
+  // Colors are selected from https://convertingcolors.com/list/avery.html.
+  // Use reddish colors for different used memories, yellowish colors for different cached memories
+  // and greenish colors for different unused memories.
+  const std::array<Color, kCGroupAndProcessMemoryTrackDimension> kCGroupAndProcessMemoryTrackColors{
+      Color(231, 68, 53, 255),   // red
+      Color(249, 96, 111, 255),  // warm red
+      Color(246, 196, 0, 255),   // orange
+      Color(87, 166, 74, 255)    // green
+  };
+  SetSeriesColors(kCGroupAndProcessMemoryTrackColors);
+
+  const std::string kValueLowerBoundLabel = "Minimum: 0 GB";
+  constexpr double kValueLowerBoundRawValue = 0.0;
+  TrySetValueLowerBound(kValueLowerBoundLabel, kValueLowerBoundRawValue);
+
+  const std::string kGameCGroupName = "game";
+  constexpr float kGameCGroupLimitGB = 7;
+  if (cgroup_name_ == kGameCGroupName) {
+    const std::string kValueUpperBoundTooltips = absl::StrFormat(
+        "<b>The memory production limit of the %s cgroup is %.2fGB</b>.<br/><br/>"
+        "<i>To launch game with the production cgroup limits, add the flag "
```

Comment:
Nit: This tooltip is not visible in the screenshot, but I suppose that it is in Orbit. Feel free to check it.

Reply:
This is included in the last screenshot I listed above https://screenshot.googleplex.com/BXTz9uhnjxFmtYN